From e7f0c46f1b75a97cce2674c4e0455a568801a919 Mon Sep 17 00:00:00 2001
From: Ryo Hashimoto <hashimoto@chromium.org>
Date: Thu, 2 May 2024 06:27:06 +0000
Subject: [PATCH] [PATCH] Fix partition_alloc buildflag generation logic for
 libchrome

- Fix buildflag_header.gni to locate the script correctly, and to remove
  the dependency on the build_config source set which doesn't exist in
  libchrome.
- Fix write_buildflag_header.py to fix the include paths in the
  generated header files.

Change-Id: I2bedffea30ecb10a61721ff0bccdc57424f8324c
---
 .../src/partition_alloc/buildflag_header.gni                  | 4 +---
 .../src/partition_alloc/write_buildflag_header.py             | 2 +-
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/base/allocator/partition_allocator/src/partition_alloc/buildflag_header.gni b/base/allocator/partition_allocator/src/partition_alloc/buildflag_header.gni
index a1c752a2bc..baae7e7a4f 100644
--- a/base/allocator/partition_allocator/src/partition_alloc/buildflag_header.gni
+++ b/base/allocator/partition_allocator/src/partition_alloc/buildflag_header.gni
@@ -79,7 +79,7 @@ _current_dir = get_path_info(".", "abspath")
 #   }
 template("pa_buildflag_header") {
   action(target_name) {
-    script = "./write_buildflag_header.py"
+    script = "${_current_dir}/write_buildflag_header.py"
 
     if (defined(invoker.header_dir)) {
       header_file = "${invoker.header_dir}/${invoker.header}"
@@ -115,7 +115,5 @@ template("pa_buildflag_header") {
                              "testonly",
                              "visibility",
                            ])
-
-    public_deps = [ "${_current_dir}:build_config" ]
   }
 }
diff --git a/base/allocator/partition_allocator/src/partition_alloc/write_buildflag_header.py b/base/allocator/partition_allocator/src/partition_alloc/write_buildflag_header.py
index e6adb7793c..ed9d7c5a32 100755
--- a/base/allocator/partition_allocator/src/partition_alloc/write_buildflag_header.py
+++ b/base/allocator/partition_allocator/src/partition_alloc/write_buildflag_header.py
@@ -86,7 +86,7 @@ def WriteHeader(options):
 
     output_file.write('\n#ifndef %s\n' % options.header_guard)
     output_file.write('#define %s\n\n' % options.header_guard)
-    output_file.write('#include "partition_alloc/buildflag.h" // IWYU pragma: export\n\n')
+    output_file.write('#include "base/allocator/partition_allocator/src/partition_alloc/buildflag.h" // IWYU pragma: export\n\n')
     # TODO(https://crbug.com/41481467) Remove dependency on chromium:
     output_file.write('#include "build/buildflag.h" // IWYU pragma: export\n\n')
 
-- 
2.45.0.rc0.197.gbae5840b3b-goog

